From 3a5234187911c09489b7f77a03999c9ec87491ac Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Mon, 21 Dec 2020 12:06:25 +0000
Subject: [PATCH 110/118] arm64: dts: meson: add i2c3/rtc nodes and vrtc alias
 to ODROID N2+

The ODROID N2+ ships with RTC hardware onboard, so enable it. Even if
there is no power cell installed this results in faster reboot times.

Also add alias entries to ensure vrtc does not claim /dev/rtc0.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../dts/amlogic/meson-g12b-odroid-n2-plus.dts   | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts
index ce1198ad34e4..80633e43f289 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts
@@ -13,6 +13,23 @@
 / {
 	compatible = "hardkernel,odroid-n2-plus", "amlogic,s922x", "amlogic,g12b";
 	model = "Hardkernel ODROID-N2Plus";
+
+	aliases {
+		rtc0 = &rtc;
+		rtc1 = &vrtc;
+	};
+};
+
+&i2c3 {
+	status = "okay";
+	pinctrl-0 = <&i2c3_sda_a_pins>, <&i2c3_sck_a_pins>;
+	pinctrl-names = "default";
+
+	rtc: rtc@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		wakeup-source;
+	};
 };
 
 &vddcpu_a {
-- 
2.31.1

